readium/navigator/src/main/java/org/readium/r2/navigator/media3/tts/TtsEngineFacade.kt

```diff
@@ -0,0 +1,98 @@
+/*
+ * Copyright 2022 Readium Foundation. All rights reserved.
+ * Use of this source code is governed by the BSD-style license
+ * available in the top-level LICENSE file of the project.
+ */
+
+package org.readium.r2.navigator.media3.tts
+
+import java.util.*
+import kotlinx.coroutines.CancellableContinuation
+import kotlinx.coroutines.ExperimentalCoroutinesApi
+import kotlinx.coroutines.suspendCancellableCoroutine
+import org.readium.r2.navigator.preferences.Configurable
+import org.readium.r2.shared.ExperimentalReadiumApi
+import org.readium.r2.shared.util.Language
+
+@ExperimentalReadiumApi
+@OptIn(ExperimentalCoroutinesApi::class)
+internal class TtsEngineFacade<S : TtsEngine.Settings, P : TtsEngine.Preferences<P>,
+    E : TtsEngine.Error, V : TtsEngine.Voice>(
+    private val engine: TtsEngine<S, P, E, V>
+) : Configurable<S, P> by engine {
+
+    init {
+        val listener = EngineListener()
+        engine.setListener(listener)
+    }
+
+    private var currentTask: UtteranceTask<E>? = null
```

Comment:
Coroutine-safety is supposed to be guaranteed by the caller which ensures that only one coroutine at once calls `speak`. This a bit tricky, but a mutex would cause a dead lock that seems to me even more trickier to prevent.